assets/src/hooks/useApiCall.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from "react"
+
+/**
+ * Return type of {@linkcode useApiCall}
+ */
+interface ApiLoadingResult<T> {
+  /**
+   * Indicates if {@linkcode useApiCall} is in the process of computing a new
+   * value
+   */
+  isLoading: boolean
+
+  /**
+   * The last value that {@linkcode useApiCall} resolved.
+   *
+   * Is `undefined` when it has not resolved or it resolves to `undefined`
+   */
+  result: T | undefined
+}
+
+/**
+ * Calls a function returning a {@linkcode Promise} and tracks the loading state
+ * and the previously resolved value.
+ *
+ * @param apiCall Function which returns a promise that is called every time it changes.
+ * @returns {ApiLoadingResult<T>}
+ */
+export const useApiCall = <T>(
+  apiCall: (abortSignal: AbortSignal) => Promise<undefined | T>
+): ApiLoadingResult<T> => {
+  const [apiResult, setApiResult] = useState<T | undefined>(undefined)
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    const controller = new AbortController()
```

Comment:
I found out this exists!
I'm not sure if this is better or should be preferred over using a simple `let cancel = false` definition, but I liked how this made the code a bit cleaner to understand, also allowing me to pass in a read only reference to this to the `apiCall` parameter as well!